### PR TITLE
Introduce full-userinfo scope for third party app

### PR DESCRIPTION
**Pull request by carmenlau**
refs #2442

### CODE REVIEW

**Comment by carmenlau** [comment]
Only include essential standard attributes (email, email_verified, phone, phone_number_verified, preferred_username) in ID token

File: pkg/auth/handler/oauth/userinfo.go [hunk 1]
```diff
@@ -30,13 +32,17 @@ type UserInfoHandler struct {
...
 	Database         *appdb.Handle
 	UserInfoProvider ProtocolUserInfoProvider
+	OAuth            *config.OAuthConfig
 }
 
 func (h *UserInfoHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	s := session.GetSession(r.Context())
+	scopes := oauth.SessionScopes(s)
+	clientID := oauth.ClientID(s)
+	client, _ := h.OAuth.GetClient(clientID)
```
Comment (louischan-oursky) [inline pkg/auth/handler/oauth/userinfo.go:42]: client could also be nil if the third party client is deleted. We now assume a nil client is IDP session. If the client is deleted, we treat it as third party with no scopes.

File: pkg/lib/oauth/oidc/id_token.go [hunk 2]
```diff
@@ -239,7 +239,7 @@ func (ti *IDTokenIssuer) PopulateUserClaims(token jwt.Token, userID string, nonP
 	return nil
 }
 
-func (ti *IDTokenIssuer) GetUserInfo(userID string) (map[string]interface{}, error) {
+func (ti *IDTokenIssuer) GetUserInfo(userID string, client *config.OAuthClientConfig, scopes []string) (map[string]interface{}, error) {
```
Comment (louischan-oursky) [inline pkg/lib/oauth/oidc/id_token.go:253]: Maybe we can introduce a data structure to represent the following states

1. IDP session
2. First party client
3. Third party client

```golang
type ClientParty string
const ClientPartyFirst ClientParty = "first_party"
const clientPartyThird ClientParty = "third_party"

type ClientLike struct {
  ClientParty ClientParty
  Scopes []string
}

var ClientLikeIDPSession = &ClientLike{
  ClientParty: ClientPartyFirst,
  Scopes: []string{FullAccessScope},
}

var ClientLikeNotFound = &ClientLike{
  ClientParty: ClientPartyThird,
}

func (s *IDPSession) ClientLike() *ClientLike {
  return ClientLikeIDPSession
}

func (s *OfflineGrant) ClientLike(c *config.OAuthConfig) *ClientLike {
  client, ok := c.GetClient(s.ClientID)
  if !ok {
    return ClientLikeNotFound
  }
  return &ClientLike{
    ClientParty: client.ClientParty(),
    Scopes: s.Scopes,
  }
}
```

We can remove `IsFirstParty()` and use `ClientParty()` instead.

GetUserInfo should takes `(userID string, client *ClientLike)`

File: pkg/auth/handler/oauth/userinfo.go (hunk 1 above)
Comment (carmenlau) [reply pkg/auth/handler/oauth/userinfo.go]: The session should become invalid if the client is deleted, so I didn't handle this case.

Updated to introduce `ClientLike` to handle the case :)

File: pkg/lib/oauth/oidc/id_token.go (hunk 2 above)
Comment (carmenlau) [reply pkg/lib/oauth/oidc/id_token.go]: Rewrite the commits to introduce `ClientLike`, and see if it looks better!

**Pull request merged**